missing language files cause 404s everywhere

I hit this in testing a pull request for dataverse-ansible, and @qqmyers relays that he hit the same bug today in his own testing: "I set the language dir and had not yet populated that dir with Bundle."

Once Ansible creates the language dir, it attempts to upload the language pack but receives: `"status": 404, "url": "http://localhost:8080/api/admin/datasetfield/loadpropertyfiles"`

Jim says:

> Possibly a catch 22 for this API call since it is trying to write to the lang dir I think. A workaround could be to write a Bundle_en.properties file (or the default machine language for other locales) - it might even be OK to have it empty (if not copy the Bundle.properties file) since I think there is logic to default back to Bundles.properties if a given key isn't found.

server.log says

```
Exception while invoking class org.glassfish.ejb.startup.EjbDeployer load method
java.lang.RuntimeException: EJB Container initialization error
        at org.glassfish.ejb.startup.EjbApplication.loadContainers(EjbApplication.java:237)
        at org.glassfish.ejb.startup.EjbDeployer.load(EjbDeployer.java:286)
        at org.glassfish.ejb.startup.EjbDeployer.load(EjbDeployer.java:104)
        at org.glassfish.internal.data.ModuleInfo.load(ModuleInfo.java:219)
        at org.glassfish.internal.data.ApplicationInfo.load(ApplicationInfo.java:336)
        at com.sun.enterprise.v3.server.ApplicationLifecycle.prepare(ApplicationLifecycle.java:570)
        at com.sun.enterprise.v3.server.ApplicationLoaderService.processApplication(ApplicationLoaderService.java:416)
        at com.sun.enterprise.v3.server.ApplicationLoaderService.postConstruct(ApplicationLoaderService.java:248)
        at org.jvnet.hk2.internal.ClazzCreator.postConstructMe(ClazzCreator.java:303)
        at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:351)
        at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:463)
        at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:281)
        at org.glassfish.hk2.runlevel.RunLevelContext.findOrCreate(RunLevelContext.java:65)
        at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:2102)
        at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:93)
        at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:67)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.oneJob(CurrentTaskFuture.java:1213)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.run(CurrentTaskFuture.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ExceptionInInitializerError
        at edu.harvard.iq.dataverse.validation.PasswordValidatorServiceBean.<init>(PasswordValidatorServiceBean.java:102)
        at edu.harvard.iq.dataverse.validation.__EJB31_Generated__PasswordValidatorServiceBean__Intf____Bean__.<init>(Unknown Source)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at com.sun.ejb.containers.BaseContainer.instantiateOptionalEJBLocalBusinessObjectImpl(BaseContainer.java:3800)
        at com.sun.ejb.containers.StatelessSessionContainer.initializeHome(StatelessSessionContainer.java:216)
        at com.sun.ejb.containers.StatelessContainerFactory.createContainer(StatelessContainerFactory.java:63)
        at org.glassfish.ejb.startup.EjbApplication.loadContainers(EjbApplication.java:225)
        ... 20 more
Caused by: java.util.MissingResourceException: Can't find bundle for base name Bundle, locale en
        at java.base/java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:2055)
        at java.base/java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1689)
        at java.base/java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1593)
        at java.base/java.util.ResourceBundle.getBundle(ResourceBundle.java:1284)
        at edu.harvard.iq.dataverse.util.BundleUtil.getResourceBundle(BundleUtil.java:99)
        at edu.harvard.iq.dataverse.util.BundleUtil.getResourceBundle(BundleUtil.java:83)
        at edu.harvard.iq.dataverse.util.BundleUtil.getStringFromBundle(BundleUtil.java:30)
        at edu.harvard.iq.dataverse.util.BundleUtil.getStringFromBundle(BundleUtil.java:26)
        at edu.harvard.iq.dataverse.validation.GoodStrengthRule.<clinit>(GoodStrengthRule.java:18)
        ... 31 more
```

To reproduce, clone https://github.com/dsd-sztaki-hu/dataverse-ansible/tree/languages and set `dataverse.language.enabled` to `true`

--- a/src/main/java/edu/harvard/iq/dataverse/util/BundleUtil.java
+++ b/src/main/java/edu/harvard/iq/dataverse/util/BundleUtil.java
@@ -93,14 +93,15 @@ public class BundleUtil {
         }
 
         if (filesRootDirectory == null || filesRootDirectory.isEmpty()) {
-            bundle = ResourceBundle.getBundle("propertyFiles/" +propertyFileName, currentLocale);
+            bundle = ResourceBundle.getBundle("propertyFiles/" + propertyFileName, currentLocale);
         } else {
             try {
-            ClassLoader loader = getClassLoader(filesRootDirectory);
-            bundle = ResourceBundle.getBundle(propertyFileName, currentLocale, loader);
+                ClassLoader loader = getClassLoader(filesRootDirectory);
+                bundle = ResourceBundle.getBundle(propertyFileName, currentLocale, loader);
             } catch (MissingResourceException mre) {
-                logger.warning("No property file named " + propertyFileName+"_" + currentLocale.getLanguage() + " found in " + filesRootDirectory +", using untranslated values");
-                bundle = ResourceBundle.getBundle("propertyFiles/" +propertyFileName, currentLocale);
+                logger.warning("No property file named " + propertyFileName + "_" + currentLocale.getLanguage()
+                        + " found in " + filesRootDirectory + ", using untranslated values");
+                bundle = ResourceBundle.getBundle("propertyFiles/" + propertyFileName, currentLocale);
             }
         }
 

--- a/src/main/java/edu/harvard/iq/dataverse/util/BundleUtil.java
+++ b/src/main/java/edu/harvard/iq/dataverse/util/BundleUtil.java
@@ -95,8 +95,13 @@ public class BundleUtil {
         if (filesRootDirectory == null || filesRootDirectory.isEmpty()) {
             bundle = ResourceBundle.getBundle("propertyFiles/" +propertyFileName, currentLocale);
         } else {
+            try {
             ClassLoader loader = getClassLoader(filesRootDirectory);
             bundle = ResourceBundle.getBundle(propertyFileName, currentLocale, loader);
+            } catch (MissingResourceException mre) {
+                logger.warning("No property file named " + propertyFileName+"_" + currentLocale.getLanguage() + " found in " + filesRootDirectory +", using untranslated values");
+                bundle = ResourceBundle.getBundle("propertyFiles/" +propertyFileName, currentLocale);
+            }
         }
 
         return bundle ;

